staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode.go

```diff
@@ -139,13 +139,31 @@ var Decode cbor.DecMode = func() cbor.DecMode {
 		// Disable default recognition of types implementing encoding.BinaryUnmarshaler,
 		// which is not recognized for JSON decoding.
 		BinaryUnmarshaler: cbor.BinaryUnmarshalerNone,
+
+		// Marshal types that implement encoding.TextMarshaler by calling their MarshalText
+		// method and encoding the result to a CBOR text string.
+		TextUnmarshaler: cbor.TextUnmarshalerTextString,
 	}.DecMode()
 	if err != nil {
 		panic(err)
 	}
 	return decode
 }()
 
+var Decode cbor.DecMode = func() cbor.DecMode {
```

Comment:
when should the unexported `decode` be used, and when should this with JSON transcoding added in be used? it's not obvious why this is split out

Reply:
Updated, please let me know if the explanation doesn't make sense.